api_v1/app/app.py

```diff
@@ -0,0 +1,282 @@
+from classes.user import User
+
+app = Flask(__name__)
+
+app.secret_key = 'younoscretet'
+
+''' creating an instance of the user '''
+user = User()
+
+''' dictionary to store our users'''
+users = {}
+
+def token_required (f):
+	""" decoding received token """
+	@wraps(f)
+	def decorate(*args, **kwargs):
+		token = None
+
+		if 'Authorization' in request.headers:
+			token = request.headers['Authorization']
+
+		if not token:
+			return jsonify({'message': 'Token is missing'}), 403
+		try:
+			payload = jwt.decode(token, app.secret_key)
+			current_user = payload['sub']
+
+		except jwt.ExpiredSignatureError:
+			return jsonify ({'message': 'Expired token!'}), 403
+
+		except jwt.InvalidTokenError:
+			return jsonify({'message': 'Invalid token!'}), 403
+
+		return f(current_user, *args, **kwargs)
+
+	return decorate
+
+
+@app.route('/')
+def display_documentation():
+	""" route displays api documentation """
+
+	return render_template('index.html')
+
+
+@app.route('/api/v1/auth/register', methods = ['POST'])
+def register():
+	""" route enables a user to register """
+
+	data = request.get_json()
+	if data:
+
+		username = data['username']
+		name_match = re.match('^[A-Za-z0-9]+$', username)
+		password = data['password']
+		password_match = re.match('^[A-Za-z0-9]{4,}$', password)
+
+		if name_match and password_match:
+			''' checking if user is not already registered '''
+
+			if username not in users.keys():
+				users[username] = password
+				return jsonify({'success': True,
+							    'message': 'Registered successfully',
+				                'data': {'Username': username}
+				                }), 201
+
+			return jsonify({'success': False,
+			                'message': 'User already registered!'}), 202
+		else:
+		    return jsonify({'success': False,
+			                'message':'Your password must be 4 or more characters, No special characters for name and password'}),403
+
+	return jsonify({'success': False,
+		            'message': 'No data was passed!'}),400
+
+
+@app.route('/api/v1/auth/login', methods = ['POST'])
+def login():
+	""" route enables a user to login """
+
+	data = request.get_json()
+	username = data['username']
+	name_match = re.match('(^[A-Za-z0-9]+$)', username)
+	password = data['password']
+	password_match = re.match('^[A-Za-z0-9]{4,}$', password)
+
+	if name_match and password_match:
+
+		#if check_password_hash(users[username], password):
+		if ( username, password ) in users.items():
+			''' checking user is already registered '''
+
+			# generate access token
+			payload = {
+			        'exp': datetime.datetime.utcnow() + datetime.timedelta( minutes = 15),
+			        'iat': datetime.datetime.utcnow(),
+			        'sub': 1
+			}
+			token = jwt.encode (
+				payload,
+				app.secret_key,
+				algorithm = 'HS256'
+				)
+
+			session['logged_in'] = True
+			return jsonify({'success': True,
+				            'message': 'Logged in successfully!',
+				            'token': token.decode('UTF-8')}), 200
```

Comment:
We don't need to again decode the token here, this should be handled by the `token_required` decorator.